Dockerfile-debian.template

```diff
@@ -73,6 +84,9 @@ RUN set -eux; \
 # "graceful stop is triggered when the SIGUSR1 signal is sent to the haproxy process"
 STOPSIGNAL SIGUSR1
 
-COPY docker-entrypoint.sh /
-ENTRYPOINT ["/docker-entrypoint.sh"]
+COPY docker-entrypoint.sh /usr/local/bin/
+RUN ln -s usr/local/bin/docker-entrypoint.sh / # backwards compat
+ENTRYPOINT ["docker-entrypoint.sh"]
+
+USER haproxy
```

Comment:
Note: It is recommended to launch HAProxy as root and to specify `user` within the configuration instead. This allows HAProxy to adjust its ulimit (e.g. the number of file descriptors) and also to chroot itself.

Reply:
Hmm, I guess we've got a difficult choice to make then - setting both of those via the container runtime is going to be strictly more secure (and a `chroot` inside a container is probably not as compelling).

Reply:
Yeah, I understand that having the container around HAProxy changes the situation a bit. I can't recommend on the "HAProxy in Docker" perspective here. I run all services in Docker, but HAProxy is very intentionally not within a container and instead deployed on the host.

I've also just verified that having a `user haproxy` in the configuration will not require any changes to the config (a `user somethingelse` will however), so this change should also be compatible with copy and paste configurations found on the Internet (unless they also put `chroot`).

I'll leave it up to you to decide what to do with the `USER` like, I just wanted to note this from a purely HAProxy perspective :smile:

Reply:
I guess that's getting tricky then. The thing is, on some platform, you can not start a container with an entrypoint that is done by `root`. EKS rejects the initial image for that reason, and I tried to have the process handled with the user, but only to start the container with `USER` could make it work.